sycl/include/sycl/info/aspects.def

```diff
@@ -19,7 +19,6 @@ __SYCL_ASPECT(ext_intel_gpu_slices, 21)
 __SYCL_ASPECT(ext_intel_gpu_subslices_per_slice, 22)
 __SYCL_ASPECT(ext_intel_gpu_eu_count_per_subslice, 23)
 __SYCL_ASPECT(ext_intel_max_mem_bandwidth, 24)
-__SYCL_ASPECT(ext_intel_mem_channel, 25)
```

Comment:
Note for reviewers: I wonder if we need to update the IDs of the aspects that follow the removed aspect. 
Some of the past PRs didn't update:
https://github.com/intel/llvm/pull/13351
So I haven't updated either.